tests/rptest/tests/topic_id_migrator.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.util import wait_until_result
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST, LoggingConfig
+from rptest.services.redpanda_installer import wait_for_num_versions, RedpandaInstaller
+from rptest.clients.kcl import RawKCL
+
+
+class TopicIDUpgradeTest(RedpandaTest):
```

Comment:
Should we also add a test that ensures that topic IDs are consistent across restarts to ensure the snapshot is working properly? I don't know if a restart is sufficient to test the snapshot.